core/controllers/editor_test.py

```diff
@@ -1530,6 +1532,285 @@ def test_email_functionality_cannot_be_used_by_non_moderators(self):
+                        'value': 200
+                    }
+                },
+                'playthrough_ids': [self.playthrough_id1],
+                'schema_version': 1,
+                'is_valid': True
+            }, {
+                'issue_type': 'MultipleIncorrectSubmissions',
+                'issue_customization_args': {
+                    'state_name': {
+                        'value': 'state_name1'
+                    },
+                    'num_times_answered_incorrectly': {
+                        'value': 7
+                    }
+                },
+                'playthrough_ids': [self.playthrough_id2],
+                'schema_version': 1,
+                'is_valid': True
+            }]
+        )
+
+    def test_fetch_issues_handler(self):
+        """Test that all issues get fetched correctly."""
+        response = self.get_json(
+            '/issuesdatahandler/%s' % (self.EXP_ID), {'exp_version': 1})
+        self.assertEqual(len(response), 2)
+        self.assertEqual(response[0]['issue_type'], 'EarlyQuit')
+        self.assertEqual(
+            response[1]['issue_type'], 'MultipleIncorrectSubmissions')
+
+    def test_invalid_issues_are_not_retrieved(self):
+        """Test that invalid issues are not retrieved."""
+        exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
+            self.EXP_ID, 1)
+        exp_issues_model.unresolved_issues[1]['is_valid'] = False
+        exp_issues_model.put()
+
+        response = self.get_json(
+            '/issuesdatahandler/%s' % (self.EXP_ID), {'exp_version': 1})
+        self.assertEqual(len(response), 1)
+        self.assertEqual(response[0]['issue_type'], 'EarlyQuit')
+
+    def test_fetch_playthrough_handler(self):
+        """Test that the playthrough gets fetched correctly."""
+        response = self.get_json(
+            '/playthroughdatahandler/%s/%s' % (
+                self.EXP_ID, self.playthrough_id1))
+        self.assertEqual(response['id'], self.playthrough_id1)
+        self.assertEqual(response['exp_id'], self.EXP_ID)
+        self.assertEqual(response['exp_version'], 1)
+        self.assertEqual(response['issue_type'], 'EarlyQuit')
+        self.assertEqual(
+            response['issue_customization_args'], {
+                'state_name': {
+                    'value': 'state_name1'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            })
+        self.assertEqual(
+            response['actions'], [{
+                'action_type': 'ExplorationStart',
+                'action_customization_args': {
+                    'state_name': {
+                        'value': 'state_name1'
+                    }
+                },
+                'schema_version': 1
+            }])
+
+
+class ResolveIssueHandlerTest(test_utils.GenericTestBase):
+    """Test the handler for resolving issues."""
+
+    EXP_ID = 'exp_id1'
+
+    def setUp(self):
```

Comment:
I see a lot of duplicate code in these `setUp` methods. Consider moving some to a common base class?

Reply:
@brianrodri That's a separate project me and @seanlip  were discussing today. In the future, we would have a common test framework for a lot of functionalities like creating and publishing an exploration, and so on. For now, we're just aiming to achieve 100% coverage.

Reply:
Resolved.